lib/iris/tests/unit/cube/test_Cube.py

```diff
@@ -1082,38 +1085,60 @@ def test_lazy(self):
         )
         _shared_utils.assert_masked_array_equal(expected_result, res_cube.data)
 
-    def test_ancillary_variables_and_cell_measures_kept(self):
+    @pytest.mark.parametrize("dataless", [True, False])
+    def test_ancillary_variables_and_cell_measures_kept(self, dataless):
+        if dataless:
+            self.cube.data = None
         res_cube = self.multi_dim_cube.rolling_window("val", self.mock_agg, 3)
         assert res_cube.ancillary_variables() == [self.ancillary_variable]
         assert res_cube.cell_measures() == [self.cell_measure]
 
-    def test_ancillary_variables_and_cell_measures_removed(self):
+    @pytest.mark.parametrize("dataless", [True, False])
+    def test_ancillary_variables_and_cell_measures_removed(self, dataless):
+        if dataless:
+            self.cube.data = None
         res_cube = self.multi_dim_cube.rolling_window("extra", self.mock_agg, 3)
         assert res_cube.ancillary_variables() == []
         assert res_cube.cell_measures() == []
 
-    def test_weights_arr(self):
+    @pytest.mark.parametrize("dataless", [True, False])
+    def test_weights_arr(self, dataless):
```

Comment:
This test doesn't assert anything for dataless cubes. Perhaps don't parametrize it for dataless?